### PR TITLE
✨ Make file uploader use field labels for file names

### PR DESCRIPTION
**Pull request by Sven65**
## Explain the changes you’ve made

In accordance with #CU-1ptbxna, I've made uploaded files in forms use the field label as a prefix for file names.

## Explain your solution

I've added an optional prop to ImageUploader which, when specified, uses that value for the file names, instead of the original file name.

As we don't have anything else to pull categories from, I'm feeding this value from the label of a FormField.

## How to test

Concrete example:

1. Checkout this branch
2. Start a form where you can upload files
3. Upload files
4. See that the file names have changed to use the label.

## Tested environments

- [] Storybook on a iOS device/simulator.
- [x] Building the Application on a iOS device/simulator.
- [] Building the Application on a Android device/simulator.

---

## Checklist for reviewers

Here are a few points that are useful for reviewers to check.

The short list of the (maybe) most important points are:

- Does this PR work as expected?
- Does the PR properly only concern a single feature/bug/thing?
- Is the code clear and concise?
- Will the changes impact existing functionality?
- Are there any linting errors (ESLint) that should be resolved?
- Do the automatic GitHub checks pass?

However it is good if a reviewer can be as thorough as possible,
within reasonable limits of course.

### Functionality

- Does this PR work as expected?
- If this is a new feature, does it add value, or is it a sign of feature-creep?
- Will the changes impact existing functionality?
- Does the changes create inconsistencies somewhere else?
- Does the code work on different platforms if applicable?
- How does the code handle if unexpected values are passed to functions etc.?

### Readability, syntax, formatting, style

- Is the code clear and concise?
- Has styling rules (Prettier) been applied to included files?
- Are functions/variables/etc. aptly named?

### Design

- Is the code properly planned and designed?
- Is code organized/separated in a logical fashion?
- Does the code follow clean code practices to a reasonable degree?
- Is the code self-documenting enough?
- Is the code in sync with existing patterns/technologies?
- Does the code reuse existing functionality where applicable?
- Is the code designed with reusability in mind?
- Does the code have hard-coded values that can be refactored?

### Patterns, idioms, tests, best practices

- Are there any linting errors (ESLint) that should be resolved?
- Is the code properly typed (TypeScript)?
- Does the code utilize language features to their full potential?

### Debugging, testing, configuring

- Is appropriate logging used?
- Have tests been added/modified to accommodate the changes?
- Is the feature sufficiently configurable in case tweaks are required in the future?

### Performance, reliability, scalability

- Is the code well optimized?
- Does the code scale?
- Does the code handle failure scenarios?

### Security

- Is there any obvious potential for exploits and security threats?
- Does code check proper authorization where applicable?

### Etiquettes

- Are commits atomic (does only one thing per commit)?
- Does the PR properly only concern a single feature/bug/thing?
- Are commit messages well-written?
- Do the automatic GitHub checks pass?

### CODE REVIEW

**Comment by helsingborgsstad** [comment]
Task linked: [CU-1ptbxna Som handläggare vill jag att filer som invånaren bifogar kommer in med tydliga filnamn för att det ska bli enklare handläggning](https://app.clickup.com/t/1ptbxna)

File: source/components/molecules/ImageUploader/ImageUploader.tsx [hunk 1]
```diff
@@ -118,13 +119,25 @@ const ImageUploader: React.FC<Props> = ({
...
+  ...rest
 }) => {
   const [choiceModalVisible, toggleModal] = useModal();
 
+  const renameImagesWithPrefix = (images: Image[], prefix: string) =>
+    images.map((image, idx) => ({
+      ...image,
+      filename: `${prefix}_${idx}`,
+    }));
+
```
Comment (karlknowit) [inline source/components/molecules/ImageUploader/ImageUploader.tsx:132]: It's a better practise to not loop in a function, but instead loop outside of it and make the function do one thing only like:

```js
const renameImageWithPrefix = (image: Image, prefix: string) => (
  {
   ...image,
   fileName: `${prefix}_${idx}`,
  }
);

if (imageNamePrefix) {
  updatedImages = updatedImages.map((image) => 
    renameImageWithPrefix(updatedImage, imageNamePrefix));
}
```

File: source/components/molecules/ImageUploader/ImageUploader.tsx [hunk 2]
```diff
@@ -118,13 +119,25 @@ const ImageUploader: React.FC<Props> = ({
...
   const addImagesToState = (newImages: Image[]) => {
-    const updatedImages =
+    let updatedImages =
       images === "" ? [...newImages] : [...images, ...newImages];
 
+    if (imageNamePrefix) {
+      updatedImages = renameImagesWithPrefix(updatedImages, imageNamePrefix);
+    }
+
```
Comment (karlknowit) [inline source/components/molecules/ImageUploader/ImageUploader.tsx:140]: **OPTIONAL**

One (optional) thing you could do is to give `imageNamePrefix` a default value (imageNamePrefix = ''), then you do not need to have the if-statement for checking if the value is set.

**Review by karlknowit — CHANGES_REQUESTED** [review]

File: source/components/molecules/ImageUploader/ImageUploader.tsx (hunk 2 above)
Comment (Sven65) [reply source/components/molecules/ImageUploader/ImageUploader.tsx:140]: If I don't have the check, and instead only use the `renameImagesWithPrefix` function, the original filenames will not be preserved when the `imageNamePrefix` is not set, but will instead be named in the pattern `_${idx}`.

File: source/components/molecules/ImageUploader/ImageUploader.tsx [hunk 3]
```diff
@@ -124,18 +124,23 @@ const ImageUploader: React.FC<Props> = ({
...
+    filename: `${prefix}_${suffix}`,
+  });
 
   const addImagesToState = (newImages: Image[]) => {
     let updatedImages =
       images === "" ? [...newImages] : [...images, ...newImages];
 
     if (imageNamePrefix) {
-      updatedImages = renameImagesWithPrefix(updatedImages, imageNamePrefix);
+      updatedImages = updatedImages.map((image, idx) =>
```
Comment (karlknowit) [inline source/components/molecules/ImageUploader/ImageUploader.tsx:141]: No abbreviations. `idx` -> `index`

File: source/components/molecules/ImageUploader/ImageUploader.tsx [hunk 4]
```diff
@@ -124,18 +124,23 @@ const ImageUploader: React.FC<Props> = ({
...
-      filename: `${prefix}_${idx}`,
-    }));
+  const renameImageWithAffix = (
+    image: Image,
+    prefix: string,
+    suffix: string
+  ) => ({
+    ...image,
+    filename: `${prefix}_${suffix}`,
+  });
```
Comment (karlknowit) [inline source/components/molecules/ImageUploader/ImageUploader.tsx:134]: The usage of `prefix` and `suffix` is not entirely correct. `suffix` and `prefix` is used to add characters before or after a base word. `prefix`-`base word`- `suffix`. So in in your case the more correct logic would be:

```js
  const renameImageWithSuffix = (
    image: Image,
    imageName: string,
    suffix: string
  ) => ({
    ...image,
    filename: `${imageName}_${suffix}`,
  });


if (...) {
  updatedImages = updatedImages.map((image, index) =>
    renameImageWithSuffix(image, imageName, index.toString())
   );
}
```

File: source/components/molecules/ImageUploader/ImageUploader.tsx [hunk 5]
```diff
@@ -124,18 +124,23 @@ const ImageUploader: React.FC<Props> = ({
...
 
   const addImagesToState = (newImages: Image[]) => {
     let updatedImages =
       images === "" ? [...newImages] : [...images, ...newImages];
 
     if (imageNamePrefix) {
```
Comment (karlknowit) [inline source/components/molecules/ImageUploader/ImageUploader.tsx:140]: Perhaps: 

`imageNamePrefix` -> `preferredImageName`

**Review by karlknowit — CHANGES_REQUESTED** [review]

**Review by karlknowit — APPROVED** [review]
Looks good!

**Pull request merged**